lib/PrintAsClang/DeclAndTypePrinter.cpp

```diff
@@ -2910,6 +2910,9 @@ static bool isEnumExposableToCxx(const ValueDecl *VD,
       if (auto *params = elementDecl->getParameterList()) {
         for (const auto *param : *params) {
           auto paramType = param->getInterfaceType();
+          // TODO: properly support exporting these optionals. rdar://131112273 
+          if (paramType->isOptional() && paramType->getOptionalObjectType()->isTypeParameter())
```

Comment:
So it's exactly the `T?` form that's a problem? Or is it any use of a generic parameter, like `Array<T>?`? If the former, this identifies it. If the latter, use `hasTypeParameter()` instead.

Reply:
> So it's exactly the `T?` form that's a problem?

Yup. We try to get the declaration for the inner type of the optional via `getNominalOrBoundGenericNominal` and we dereferenced the resulting null pointer in the `T?` case.